src/typechecker.erl

```diff
@@ -2232,8 +2232,10 @@ do_type_check_expr(Env, {'try', _, Block, CaseCs, CatchCs, AfterBlock}) ->
 
 %% Maybe - value-based error handling expression
 %% See https://www.erlang.org/eeps/eep-0049
-do_type_check_expr(_Env, {'maybe', Anno, [{maybe_match, _, _LHS, _RHS}]} = MaybeExpr) ->
-    erlang:throw({unsupported_expression, Anno, MaybeExpr}).
+do_type_check_expr(_Env, {'maybe', _, _}) ->
```

Comment:
I think that equivalent code should be also added to `do_type_check_expr_in`.

Reply:
Good point, added. In the case of `do_type_check_expr_in`, though, it's not possible to assume `any()`, since the operation is checking (as in _checking_ vs _inference_ in bidirectional typing), so it would certainly lead to false positives. I think it's better to skip such functions altogether for now.